### PR TITLE
Invoice APIs

### PR DESCRIPTION
**Pull request by 201560543**
- date range filter util added to apply filter on objects
-invoice line items can be fetched by passing invoice header id

### CODE REVIEW

File: app/web/api/api_filters.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
+from datetime import datetime
+
+
+def date_range_filter(obj, filter_key ,start_date=None, end_date=None):
+    """
+        Utility function to get the date ranger filter for the passed query object.
+    """
+    if start_date and end_date:
+        return [item for item in obj if (
+                (item[filter_key] >= datetime.strptime(start_date, '%Y-%m-%d').date()) and (
```
Comment (PansulBhatt) [inline app/web/api/api_filters.py:10]: Push this conversion of datetime of start data and end_date to the top.

Comment (PansulBhatt) [reply app/web/api/api_filters.py:10]: I would like us to be a bit careful here. With date conversions we should be careful with UTC formats and timezone issues. I don't think it matters for now but a good thing to be cautious about

File: app/web/api/invoice_line_item_routes.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+        options=dict(
+            bind=db_engine
+        )
+    )
+    InvLineItemTbl = InvoiceLineItemTable(db_engine)
+    invoice_header_id = request.args.get('invoice_id')
+    if invoice_header_id is None:
+        abort(400, description="Invoice header is required in the query strings.")
+    results = memsql_session.query(InvLineItemTbl.table) \
+        .filter_by(invoice_header_id=invoice_header_id).all()
```
Comment (PansulBhatt) [inline app/web/api/invoice_line_item_routes.py:23]: Do we need .all() after .filter_by()

File: app/web/api/api_filters.py (hunk 1 above)
Comment (201560543) [reply app/web/api/api_filters.py:10]: in case start_date or end_date is None DateTime conversion will throw an error that's why I kept it within scope

**Pull request merged**